### DISCUSSION TITLE
How to use MathJax and Giscus with privacy plugin

### DISCUSSION THREAD

**Discussion by ZhiyuanChen** [squidfunk/mkdocs-material#3641] [Q&A] [answered]
Hi, 

`privacy` plugin will download everything to local, while it does not download referenced `js` in [MathJax of ](https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js?config=TeX-AMS-MML_HTMLorMML), it also makes `Giscus` unusable. Is there a workaround?

**Comment by squidfunk** [comment, marked as answer]
Thanks for asking. I've checked MathJax and yes, it [does not correctly work due to the use of non-qualified URLs](https://squidfunk.github.io/mkdocs-material/setup/ensuring-data-privacy/#caching) to resolve `*.js` and `*.woff` files. Thus, for MathJax, the solution is to download the JavaScript and additional assets manually. Sadly this can not be fixed, because we can't execute the JavaScript to generate the full URLs during the build process.

Please open an issue for the problem with Giscus, describing what does not work correctly. The solution is likely to exclude Giscus from processing with the privacy plugin. I'm thinking of adding exclude patterns, so this can be configured via `mkdocs.yml`.

__Edit__: it appears that self-hosting Giscus is possible, but requires more than just copying assets:
https://github.com/giscus/giscus/blob/main/SELF-HOSTING.md

**Reply by ZhiyuanChen** [reply]
The issue with `mathjax` is very diasppointing, it is a huge js and makes the site much slower if move to local as there would be no cdn for this. 

Ths issue of Giscus is at https://github.com/squidfunk/mkdocs-material/issues/3662

**Reply by squidfunk** [reply]
I understand and am sorry for that. However, it is not fixable from our side as I already explained. Thanks for creating the Giscus issue, I'll look into it asap. The idea is to add an exclusion list, so you could add MathJax and Giscus.

**Reply by ZhiyuanChen** [reply]
Thank you so much!
